api/app/src/providers/fitbit.ts

```diff
@@ -332,4 +336,67 @@ export class Fitbit extends Provider implements OAuth2 {
       }
     );
   }
+
+  async postAuth(token: string, userId?: string) {
+    const accessToken = JSON.parse(token).access_token;
+
+    const scopes = JSON.parse(token).scope;
+
+    const subscriptionTypes: Record<FitbitScopes, string> = {
+      [FitbitScopes.activity]: "activities",
+      [FitbitScopes.nutrition]: "foods",
+      [FitbitScopes.weight]: "body",
+      [FitbitScopes.sleep]: "sleep",
+    };
+
+    const subscriptionId = userId;
+
+    if (this.allRequiredScopesIncluded(scopes)) {
+      const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/apiSubscriptions/${subscriptionId}.json`;
+      this.createSubscription(subscriptionUrl, accessToken);
+      console.log("All scopes included. User subscribed to all Fitbit WH collectionTypes.");
+    } else {
+      for (const scope of Object.keys(subscriptionTypes)) {
+        if (scopes.includes(scope)) {
+          const subscriptionType = subscriptionTypes[scope as FitbitScopes];
+          const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/${subscriptionType}/apiSubscriptions/${subscriptionId}${subscriptionType}.json`;
+          await this.createSubscription(subscriptionUrl, accessToken);
+        }
+      }
+    }
+
+    // TODO: userRevokedAccess. Revoking the token makes it so the webhook do not trigger anything on our end. So, perhaps, this isn't necessary at all.
+    // https://github.com/metriport/metriport/issues/652
+    // const userRevokedAccessUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/userRevokedAccess/apiSubscriptions/${subscriptionId}sleep.json`;
+    // await this.createSubscription(userRevokedAccessUrl, accessToken);
+  }
+
+  // Checks if all of the required scopes were authorized by the user
+  allRequiredScopesIncluded(userScopes: string): boolean {
+    for (const fitbitScope of fullSubscriptionRequiredScopes) {
+      if (!userScopes.includes(fitbitScope)) return false;
+    }
+    return true;
```

Comment:
Also NABD, but also so you know about it: Lodash has a lot of goodies we can use in situations like this, worth the time to take a look at their docs.

In this case, we could use `intersection(...).length` - [source](https://lodash.com/docs/4.17.15#intersection)

Reply:
oh, interesting! thx for letting me know!